gensim/models/keyedvectors.py

```diff
@@ -1722,7 +1722,10 @@ def vectors_for_all(self, keys: Iterable) -> 'KeyedVectors':
             Vectors for all the given keys.
 
         """
-        vocabulary = sorted(set(filter(lambda key: key in self, keys)))
+        if isinstance(keys, Dictionary):
+            vocabulary = keys.token2id
+        else:
+            vocabulary = sorted(set(filter(lambda key: key in self, keys)))
```

Comment:
Avoid `filter` and `map`, use comprehension (idiomatic Python).

Reply:
Thank you for the feedback. As per our discussion with @gojomo in https://github.com/RaRe-Technologies/gensim/pull/3157#issuecomment-848081847, I have removed sorting and replaced this part with deduplication:
```
>>> from collections import OrderedDict
>>> list(OrderedDict.fromkeys([3, 2, 2, 3, 2, 1]))
[3, 2, 1]
```
This is still not too idiomatic, but very succint. Perhaps a comment `  # deduplicates keys` would make a maintainer's life easier.

Reply:
Is this solution OK for you?